### PR TITLE
[freeboxos] Fix start/stop audio sink

### LINKED ISSUE #17208: [freeboxos] Audio sink not created
The audio sink is not created in case the player was OFF when the thing was initialized and capabilities were not set (previously) in the thing properties (that is the systematic case when the thing is defined in a thing file).

Additionally, the video capability should be checked instead of the audio capability.

### PR DESCRIPTION
**Pull request by lolodomo**
Fix #17208

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Review by clinique — APPROVED** [review]
LGTM

File: bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/ApiConsumerHandler.java [hunk 1]
```diff
@@ -87,40 +87,62 @@ private void initializeOnceBridgeOnline(FreeboxOsHandler bridgeHandler) {
...
     }
 
+    private void startAudioSink(Receiver receiver) {
+        FreeboxOsHandler bridgeHandler = checkBridgeHandler();
+        // Only video and photo is supported by the API so use VIDEO capability for audio
+        Boolean isAudioReceiver = receiver.capabilities().get(MediaType.VIDEO);
+        if (reg == null && bridgeHandler != null && isAudioReceiver != null && isAudioReceiver.booleanValue()) {
+            ApiConsumerConfiguration config = getConfig().as(ApiConsumerConfiguration.class);
+            String callbackURL = bridgeHandler.getCallbackURL();
+            if (!config.password.isEmpty() || !receiver.passwordProtected()) {
```
Comment (lsiepel) [inline bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/ApiConsumerHandler.java:124]:
```suggestion
            if (!config.password.isBlank() || !receiver.passwordProtected()) {
```

File: bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/ApiConsumerHandler.java [hunk 2]
```diff
@@ -87,40 +87,62 @@ private void initializeOnceBridgeOnline(FreeboxOsHandler bridgeHandler) {
...
+                reg = bridgeHandler.getBundleContext()
+                        .registerService(
+                                AudioSink.class.getName(), new AirMediaSink(this, bridgeHandler.getAudioHTTPServer(),
+                                        callbackURL, receiver.name(), config.password, config.acceptAllMp3),
+                                new Hashtable<>());
+                logger.debug("Audio sink registered for {}.", receiver.name());
+            } else {
+                logger.info("A password needs to be configured to enable Air Media capability.");
```
Comment (lsiepel) [inline bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/ApiConsumerHandler.java:132]: I would suggest warn level

**Review by lsiepel — CHANGES_REQUESTED** [review]
Minor comments, otherwise LGTM

File: bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/ApiConsumerHandler.java (hunk 1 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/ApiConsumerHandler.java:124]: This is @clinique original code and it is correct. Your proposal would exclude a password containing only spaces. I agree this is very improbable and I see no reason to exclude it.

File: bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/ApiConsumerHandler.java (hunk 2 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/ApiConsumerHandler.java:132]: This is original code but I agree with you.

Comment (lolodomo) [reply bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/ApiConsumerHandler.java]: Changed

File: bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/ApiConsumerHandler.java (hunk 1 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/ApiConsumerHandler.java:124]: For almost all password checks when isEmpty is used, isBlank is meant to be used. But ok. If we want to allow all whitespace passwords (yikes) I’ll merge this. 😁

**Review by lsiepel — APPROVED** [review]
LGTM

**Pull request merged**